modules/tinymce/src/plugins/media/test/ts/browser/IsCachedResponseTest.ts

```diff
@@ -1,69 +1,47 @@
-
-  Theme();
-  MediaPlugin();
-
-  const sWaitForAndAssertNotification = (expected: string) => {
-    return Chain.asStep(TinyDom.fromDom(document.body), [
-      UiFinder.cWaitFor('Could not find notification', 'div.tox-notification__body'),
-      Chain.mapper(Html.get),
-      Assertions.cAssertHtml('Plugin list html does not match', expected)
-    ]);
-  };
-
-  TinyLoader.setupLight((editor, onSuccess, onFailure) => {
-    const tinyUi = TinyUi(editor);
-    const tinyApis = TinyApis(editor);
-
-    Pipeline.async({}, [
-      tinyApis.sFocus(),
-      Log.stepsAsStep('TBA', 'Media: test cached response', [
-        tinyUi.sClickOnToolbar('click media button', 'button[aria-label="Insert/edit media"]'),
-        Chain.asStep({}, [
-          Chain.fromParent(
-            tinyUi.cWaitForPopup('wait for media dialog', 'div[role="dialog"]'), [
-              Chain.fromChains([
-                Utils.cSetSourceInput(tinyUi, 'test'),
-                Utils.cFakeEvent('paste'),
-                Chain.wait(0) // wait is needed because paste is triggered async
-              ]),
-              Chain.runStepsOnValue(() => [ Utils.sAssertEmbedData(tinyUi, '<div>x</div>') ]),
-              Chain.fromChains([
-                Utils.cSetSourceInput(tinyUi, 'XXX')
-              ]),
-              Chain.fromChains([
-                UiFinder.cFindIn(Utils.selectors.saveButton),
-                Mouse.cClick
-              ])
-            ]
-          )
-        ]),
-
-        sWaitForAndAssertNotification('<p>Media embed handler threw unknown error.</p>'),
-        tinyApis.sAssertContent('')
-      ])
-    ], onSuccess, onFailure);
-  }, {
-    plugins: 'media',
+describe('browser.tinymce.plugins.media.IsCachedResponseTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    plugins: [ 'media' ],
     toolbar: 'media',
-    theme: 'silver',
-    base_url: '/project/tinymce/js/tinymce',
     media_url_resolver: (data, resolve, reject) => {
       if (data.url === 'test') {
         resolve({
           html: '<div>x</div>' });
       } else {
         reject('error');
       }
-    }
-  }, success, failure);
+    },
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin, Theme ], true);
+
+  const pWaitForAndAssertNotification = async (expected: string) => {
+    const notification = await UiFinder.pWaitFor('Could not find notification', SugarBody.body(), 'div.tox-notification__body') as SugarElement<HTMLElement>;
+    const html = Html.get(notification);
+    Assertions.assertHtml('Plugin list html does not match', expected, html);
+  };
+
+  it('TBA: test cached response', async () => {
+    const editor = hook.editor();
+    TinyUiActions.clickOnToolbar(editor, 'button[aria-label="Insert/edit media"]');
+    await TinyUiActions.pWaitForDialog(editor);
+    const input = await Utils.pSetSourceInput(editor, 'test');
+    Utils.fakeEvent(input, 'paste');
+    await Waiter.pWait(0); // wait is needed because paste is triggered async
```

Comment:
Might be nothing, but this worries me a little. The code that this was added to the middle of was already async, on account of it using chains, and they still added a wait to the middle of it anyway? If the delay was just the difference between the next promise event loop vs the next setTimeout event loop then this translates fine — but if it was added for any other reason we might be better off wrapping that `pAssertEmbedData` in a `tryUntil` call somehow

Reply:
My guess is whoever wrote this wasn't aware Chains/Steps were actually async. Either way, this does specifically just need a "run in the next loop check".

Also worth noting this is really outside the scope of the conversion.

Reply:
Actually, thinking more `Step` and `Chain` would use `Promise.resolve()` which is a micro-task and therefore wouldn't guarantee to run in the next event loop. That may explain why it was there previously.